InterGalacticCurrencyConvertor/src/test/java/DataReaderTest.java

```diff
@@ -0,0 +1,165 @@
+package test.java;
+
+import static org.junit.Assert.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+import main.java.DataProcessResult;
+import main.java.DataReader;
+
+import org.junit.Test;
+
+/**
+ * Stores test data : the input and the expected output;
+ */
+class Data {
+	String mDataLine;
+	DataProcessResult mExpected;
+
+	Data(String inLine, String inResult, Boolean inSuccess) {
+		mDataLine = inLine;
+		mExpected = new DataProcessResult(inResult, inSuccess);
+	}
+}
+/**
+ * Does an integration test of the whole program.
+ */
+public class DataReaderTest {
+	private ArrayList<Data> mDataStore;
+	private DataReader mReader;
+
+	public DataReaderTest() {
+		ByteArrayInputStream vByteStream = new ByteArrayInputStream("".getBytes());
+		mReader = new DataReader(new InputStreamReader(vByteStream));
+		mDataStore = new ArrayList<>(12);
+		createTestData();
+	}
+
+	private void createTestData() {
```

Comment:
Break up the tests into separate methods - each testing a different set of scenarios. So each group below could be a separate test method.

Reply:
updated